az-hadoop-jobtype-plugin/src/main/java/azkaban/jobtype/HadoopProxy.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2019 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package azkaban.jobtype;
+
+import static org.apache.hadoop.security.UserGroupInformation.HADOOP_TOKEN_FILE_LOCATION;
+
+import java.io.File;
+import org.apache.log4j.Logger;
+import azkaban.flow.CommonJobProperties;
+import azkaban.security.commons.HadoopSecurityManager;
+import azkaban.security.commons.HadoopSecurityManagerException;
+import azkaban.utils.Props;
+
+
+/**
+ * Hadoop Proxy
+ */
+public class HadoopProxy {
+
+  private HadoopSecurityManager hadoopSecurityManager;
+  private boolean shouldProxy = false;
+  private boolean obtainTokens = false;
+  private String userToProxy = null;
+  private File tokenFile = null;
+
+  public HadoopProxy() {
+  }
+
+  public boolean isProxyEnabled() {
```

Comment:
can this class be consolidated with HadoopJobUtils?

Reply:
Yes. We can review the function in HadoopJobUtils and removed it out. For instance, cancelHadoopTokens is not referred right now. By default, we may be able to remove that.
But, I will recommend to address that in followup Pull Request instead of this one.